ptf/saidebugcounters.py

```diff
@@ -0,0 +1,2194 @@
+            if dc_oid1 != 0:
+                sai_thrift_remove_debug_counter(self.client, dc_oid1)
+            if dc_oid2 != 0:
+                sai_thrift_remove_debug_counter(self.client, dc_oid2)
+            if dc_oid3 != 0:
+                sai_thrift_remove_debug_counter(self.client, dc_oid3)
+
+
+class SwitchMultiDebugCounters(BaseDebugCounterClass):
+    ''' Switch multi Debug Counter test. '''
+
+    def runTest(self):
+        dc_oid1 = 0
+        dc_oid2 = 0
+        dc_oid3 = 0
+        try:
+            drop_list1 = [SAI_IN_DROP_REASON_IP_HEADER_ERROR,
+                          SAI_IN_DROP_REASON_L2_ANY]
+            dc_oid1 = self.createDebugCounter(
+                SAI_DEBUG_COUNTER_TYPE_SWITCH_IN_DROP_REASONS,
+                drop_list1)
+            self.assertTrue(dc_oid1 != 0, "Failed to Create DebugCounter")
+            self.assertTrue(self.verifyDebugCounterDropList(
+                dc_oid1, drop_list1))
+
+            drop_list2 = [SAI_IN_DROP_REASON_L2_ANY]
+            dc_oid2 = self.createDebugCounter(
+                SAI_DEBUG_COUNTER_TYPE_SWITCH_IN_DROP_REASONS,
+                drop_list2)
+            self.assertTrue(dc_oid2 != 0, "Failed to Create DebugCounter")
+            self.assertTrue(self.verifyDebugCounterDropList(
+                dc_oid2,
+                drop_list2))
+
+            if (self.isInDropReasonSupported(
+                    [SAI_IN_DROP_REASON_SMAC_EQUALS_DMAC])):
+                drop_list3 = [SAI_IN_DROP_REASON_SMAC_EQUALS_DMAC]
+                dc_oid3 = self.createDebugCounter(
+                    SAI_DEBUG_COUNTER_TYPE_SWITCH_IN_DROP_REASONS,
+                    drop_list3)
+                self.assertTrue(dc_oid3 != 0, "Failed to Create DebugCounter")
+                self.assertTrue(self.verifyDebugCounterDropList(
+                    dc_oid3,
+                    drop_list3))
+
+                self.testSwitchDebugCounter(
+                    dc_oid3,
+                    drop_list3,
+                    [self.smac_equals_dmac_pkt])
+                self.testSwitchDebugCounter(
+                    dc_oid1,
+                    drop_list2,
+                    [self.vlan_discard_pkt, self.zero_smac_pkt, self.ihl_pkt])
+            else:
+                self.testSwitchDebugCounter(
+                    dc_oid1,
+                    drop_list2,
+                    [self.vlan_discard_pkt, self.zero_smac_pkt, self.ihl_pkt])
+                self.testSwitchDebugCounter(
+                    dc_oid1,
+                    drop_list2,
+                    [self.vlan_discard_pkt, self.zero_smac_pkt])
+        finally:
+            if dc_oid1 != 0:
+                sai_thrift_remove_debug_counter(self.client, dc_oid1)
+            if dc_oid2 != 0:
+                sai_thrift_remove_debug_counter(self.client, dc_oid2)
+            if dc_oid3 != 0:
+                sai_thrift_remove_debug_counter(self.client, dc_oid3)
+
+
+class PortDropDIPLinkLocalTest(BaseDebugCounterClass):
```

Comment:
Change name: PortDropDipLinkLocalTest ?

Should we merge this and the following classes to have the same set up? I mean especially this:
self.createRouter()
self.createNeighbor(...)
self.createRoute(...)

Reply:
Looks good to me, I don't want to refactor the tests.